llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -551,8 +553,18 @@ void AArch64AsmPrinter::LowerKCFI_CHECK(const MachineInstr &MI) {
 void AArch64AsmPrinter::LowerHWASAN_CHECK_MEMACCESS(const MachineInstr &MI) {
   Register Reg = MI.getOperand(0).getReg();
   bool IsShort =
-      MI.getOpcode() == AArch64::HWASAN_CHECK_MEMACCESS_SHORTGRANULES;
+      ((MI.getOpcode() == AArch64::HWASAN_CHECK_MEMACCESS_SHORTGRANULES) ||
+       (MI.getOpcode() ==
+        AArch64::HWASAN_CHECK_MEMACCESS_SHORTGRANULES_FIXEDSHADOW));
   uint32_t AccessInfo = MI.getOperand(1).getImm();
+
+  if ((MI.getOpcode() == AArch64::HWASAN_CHECK_MEMACCESS_FIXEDSHADOW) ||
```

Comment:
optional: we  can make this the condition for L641 below, that would maybe be a bit less indirect?

Reply:
I guess it should be possible to have IR with .fixedshadow and call llc without parameter.
This is not compiler bug, but rather user error.
So we need either diagnostics, or reasonable fallback.

`getFixedShadowBase().value_or(0)` as fallback is LGTM

Reply:
for diagnostics see `report_fatal_error` in the file

Reply:
> optional: we can make this the condition for L641 below, that would maybe be a bit less indirect?

Done 

> getFixedShadowBase().value_or(0) as fallback is LGTM

Done.

Thanks Florian and Vitaly!